synapse/res/templates/notif.html

```diff
@@ -1,47 +1,47 @@
-{% for message in notif.messages %}
+{%- for message in notif.messages %}
```

Comment:
So `%-` removes space generated by each command line? (I'm reading [the docs](https://jinja.palletsprojects.com/en/2.11.x/templates/#whitespace-control) and don't see a specific explanation of `%-`).

And instead we could set `trim_blocks` and `lstrip_blocks` options to `True` to get the same behaviour? Sounds like something we might consider, as I'd be surprised to see which templates would *want* whitespace to be generated by block lines. I imagine they're probably the exception rather than the norm.

It's probably fine to leave that decision out of this PR though.

Reply:
For some reason the documentation around this is super confusing, but pretty much any line that has a `{% ... %}` on it ends up as a blank line (or a line with just whitespace) with the standard default settings. Switching to `{%-` will remove the whitespace before that block and `-%}` will remove whitespace after a block. The relevant doc is:

> You can also strip whitespace in templates by hand. If you add a minus sign (`-`) to the start or end of a block (e.g. a For tag), a comment, or a variable expression, the whitespaces before or after that block will be removed:

But I don't find that super clear. 🤷 

My only concern with enabling those is if someone customized templates, the whitespace is going to change on them. I'm not sure if this would really *break* anything, but we'd have to play with it a bit. For HTML templates it is likely find since whitespace gets combined (generally), but for text templates it has more implications.

I'm OK backing out this change and putting it into a separate PR for discussion if you'd like.

Reply:
I took a look for other templates that were potentially doing the same thing and didn't see any. I think in the future if we find ourselves needing to do this for multiple files then perhaps just adding another option to `read_templates` makes sense.

Does that sound good to you?

Reply:
Yeah, that sounds reasonable!